docs/content/preview/contribute/docs/widgets-and-shortcodes.md

```diff
@@ -0,0 +1,231 @@
+
+### includeCode
+
+Because it doesn't make its own code block, you can use this shortcode to build a code block from several sources.
+
+The base path is `/docs/static/`.
+
+This shortcode strips trailing whitespace from the input file.
+
+**Call `includeCode`** in a fenced code block:
+
+````markdown
+```sql
+{{%/* includeCode file="code-samples/include.sql" */%}}
+```
+````
+
+**To nest the code block**, tell the shortcode how many spaces to indent:
+
+````markdown
+1. To do this thing, use this code:
+
+    ```sql
+    {{%/* includeCode file="code-samples/include.sql" spaces=4 */%}}
+    ```
+````
+
+**To specify highlighting options**, do so on the fenced code block. This is a Hugo feature, not part of the shortcode. For example, add a highlight to lines 1 and 7-10:
+
+````markdown
+```sql {hl_lines=[1,"7-10"]}
+{{%/* includeCode file="code-samples/include.sql" */%}}
+```
+````
+
+{{< tip >}}
+For more information on highlight options, see the [Hugo docs](https://gohugo.io/content-management/syntax-highlighting/#highlighting-in-code-fences)
+{{< /tip >}}
+
+### includeFile
+
+The `includeFile` shortcode infers the code language from the filename extension (or `output` if there's no extension) and creates its own code block.
+
+The base path is `/docs/static/`.
+
+This shortcode strips trailing whitespace from the input file.
+
+**Call `includeFile`** on a line of its own:
+
+```go
+{{</* includeFile file="code-samples/include.sql" */>}}
+```
+
+**To nest the code block**, indent the shortcode:
+
+```go
+    {{</* includeFile file="code-samples/include.sql" */>}}
+```
+
+**To specify a code language** and override the default:
+
+```go
+{{</* includeFile file="code-samples/include.sql" lang="sql" */>}}
+```
+
+**To specify highlighting options**:
+
+```go
+{{</* includeFile file="code-samples/include.sql" hl_options="hl_lines=1 7-10" */>}}
+```
+
+{{< warning title="Fenced blocks are different" >}}
+
+CAREFUL! `hl_lines` takes a different form here than when you're specifying it on a fenced block: no comma, no quotes: `hl_options="hl_lines=1 7-10"`
+
+For more information on highlight options: <https://gohugo.io/content-management/syntax-highlighting/#highlight-shortcode>
+
+{{< /warning >}}
+
+### includeMarkdown
+
+Inserts the contents of a markdown file, rendered as part of the calling page. We use this primarily for [syntax diagrams](../syntax-diagrams/).
+
+## Other shortcodes
+
+Note that the way you invoke a shortcode matters. Use angle brackets (< and >) to output plain text, and percent signs (%) to have Hugo render the output as Markdown.
+
+For example, use `{{</* slack-invite */>}}` to output a bare URL for inclusion in a Markdown-style link, such as `[YugabyteDB Community Slack]({{</* slack-invite */>}})`, and use `{{%/* slack-invite */%}}` to output a clickable URL ({{% slack-invite %}}).
```

Comment:
That clickable URL doesn't look quite right in the preview

Reply:
Fixed the paragraph up a bit...